lib/agent.js

```diff
@@ -433,15 +434,19 @@ PolicyAgent.prototype.shield = function (shield) {
             .catch(function (err) {
                 self.logger.info('PolicyAgent#shield: evaluation error (%s)', err.message);
                 // handle response if unhandled
-                if (!res.headersSent) {
-                    httpUtils.sendResponse(res, err.statusCode, self.errorTemplate({
-                        status: err.statusCode,
-                        message: err.message,
-                        details: err.stack,
-                        pkg: pkg
-                    }), {
-                        'Content-Type': 'text/html'
-                    });
+                if(this.options.letClientHandleDenial){
```

Comment:
I think the 'this' here should be self! Using a this doesn't work due to shield evaluate being async and called when a promise resolves.